### ISSUE TITLE
Post type being indexed when it shouldn't be

### ISSUE THREAD

**Issue by jonathanstegall** [10up/ElasticPress#1133]
Have you searched for similar issues before submitting this one?

Yes

Is this a bug, question or feature request?

Possibly a bug

Describe the issue you encountered:

I have a filter like this:

```
add_filter( 'ep_indexable_post_types', 'indexable_post_types' );
function indexable_post_types( $post_types ) {
	$non_indexable_types = array(
		'partner_offer' => 'partner_offer',
		'wp_log'        => 'wp_log',
		'message'       => 'message',
	);
	$post_types          = array_diff_assoc( $post_types, $non_indexable_types );
	error_log( 'types is ' . print_r( $post_types, true ) );
	return $post_types;
}
```

The log result here is all of our post types except the three in the array above.

However, I'm noticing that when I search, there are `wp_log` post types in my search results. I believe these are all newer posts - that is, our index seems to have skipped the pre-existing log posts, but as new ones are created it seems to be indexing them.

Current WordPress version:

4.9.8

Current ElasticPress version:

2.6.1

Current Elasticsearch version:

Locally I'm running 6.2.4, and remotely I'm running 6.3.1. This behavior happens in both environments.

Where do you host your Elasticsearch server:

We have a single Linux server.

Other plugins installed (WooCommerce, Simple Redirect Manager, etc..):

We do have a significant number of other plugins. The wp_log post type is created by the [WP Logging](https://pippinsplugins.com/wp-logging/) library though.

Steps to reproduce:

1. Enable Search and Protected Content in ElasticPress settings.
2. Filter the indexable post types to remove a type.
3. Create new records of that type.
4. These records show up in search results.

Screenshots, if needed:

**Comment by brandwaffle** [comment]
@jonathanstegall thanks for reporting this!

I did a little digging and it looks like the Protected Content module does a get_post_types() call during a filter that it runs https://github.com/10up/ElasticPress/blob/a24d36ad4ea02ca971e5192b7160b806a7947411/features/protected-content/protected-content.php#L32. If your filter runs before this one, that means that Protected Content is getting the list of all public post types (including wp_log) and essentially "resetting" the indexable post types regardless of what earlier filters are doing.

If this is the case, it's definitely a bug. Can you try firing your filter later than the Protected Content one (set a priority of 11 or higher)? That should temporarily resolve the issue, and we'll get started on a bugfix.

**Comment by jonathanstegall** [comment]
@brandwaffle thanks for your help!

Here's what I tried:

Update my filter so it has a priority of 11, and manually create a log entry (I called it update wordpress user jonathan 1). I then did a search for wordpress user jonathan and that entry did not come up in the results.

To make sure that was working, I removed the filter and created a new entry called update wordpress user jonathan 2, and I ran the search again. The new entry showed up in the results that time.

I hope that helps! Thanks for the tip, in any case. I'll implement this on our end.
